lib/go-tc/steering.go

```diff
@@ -0,0 +1,41 @@
+package tc
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+type Steering struct {
+	DeliveryService DeliveryServiceName      `json:"deliveryService"`
+	ClientSteering  bool                     `json:"clientSteering"`
+	Targets         []SteeringSteeringTarget `json:"targets"`
+	Filters         []SteeringFilter         `json:"filters"`
+}
+
+type SteeringFilter struct {
+	DeliveryService DeliveryServiceName `json:"deliveryService"`
+	Pattern         string              `json:"pattern"`
+}
+
+type SteeringSteeringTarget struct {
+	Order           int64               `json:"order"`
```

Comment:
I'm still kinda worried about these being 64-bit ints when TR parses this JSON into 32-bit ints. Do we need them to be 64-bit?

Reply:
Ok, I agree. I was thinking validation would be better, if the Router can't accept it. But I just verified, Go fails to unmarshal numbers too big with `json: cannot unmarshal number 2147483648 into Go struct field Foo.Bar of type int32`. So using an int32 is just as good as validation; better, because we don't have to write it.

I'll change it.

Reply:
Done